walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  private static final Collection<Event> NO_EVENTS = Collections.emptySet();
+  private static final Collection<String> NO_ATTENDEES = Collections.emptySet();
+
+  // Some people that we can use in our tests.
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+
+  // All dates are the first day of the year 2020.
+  public static final int START_OF_DAY = TimeRange.getTimeInMinutes(0, 0);
+  public static final int END_OF_DAY = TimeRange.getTimeInMinutes(23, 59);
+  private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 0);
+  private static final int TIME_0830AM = TimeRange.getTimeInMinutes(8, 30);
+  private static final int TIME_0900AM = TimeRange.getTimeInMinutes(9, 0);
+  private static final int TIME_0930AM = TimeRange.getTimeInMinutes(9, 30);
+  private static final int TIME_1000AM = TimeRange.getTimeInMinutes(10, 0);
+  private static final int TIME_1100AM = TimeRange.getTimeInMinutes(11, 00);
+
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_60_MINUTES = 60;
+  private static final int DURATION_90_MINUTES = 90;
+  private static final int DURATION_1_HOUR = 60;
+  private static final int DURATION_2_HOUR = 120;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    int counter = 0;
+    int maxHours = TimeRange.WHOLE_DAY.duration();
+    long requestedDuration = request.getDuration();
+    int startTime1 = -1;
+    int endTime1 = -1;
+    int startTime2 = -1;
+    int endTime2 = -1;
+    Collection<String> people = null;
+
+    if(request.getAttendees().isEmpty()){           //if there are no attendees in the request
```

Comment:
Fun trick to make your code more readable:

If, else if, else and nested if statements have a lot of cognitive load. As I get further down the chain of else ifs, the reader needs to keep track of what we've previously excluded.

We know the first three conditions are independent of each other because we could run them in any order and still get the same results. Here's them written as independent if statements.

```
if (request.getAttendees().isEmpty() || events.isEmpty()) {
  return Arrays.asList(TimeRange.WHOLE_DAY);
}

if (request.getDuration() > maxHours) {
  return Arrays.asList();
}
```

This allows you to unnest the main algorithm so it's easier to understand.

Here's more info about this topic: https://dzone.com/articles/code-smells-if-statements

Reply:
Smart! I'll look into that. It won't be on my next commit because I don't have time to go over it now, but it's a smart idea, thanks!